test/brsTypes/Array.test.js

```diff
@@ -39,4 +39,46 @@ describe("Array", () => {
             );
         });
     });
+
+    describe("get", () => {
+        it("returns values from in-bounds indexes", () => {
+            let a = new BrsString("a");
+            let b = new BrsString("b");
+            let c = new BrsString("c");
+
+            let arr = new BrsArray([a, b, c]);
+
+            expect(arr.get(new Int32(0))).toBe(a);
+            expect(arr.get(new Int32(2))).toBe(c);
+        });
+
+        it("returns invalid for out-of-bounds indexes", () => {
+            let arr = new BrsArray([]);
+
+            expect(arr.get(new Int32(555))).toBe(BrsInvalid.Instance);
+        });
+    });
+
+    describe("set", () => {
+        it("sets values at in-bounds indexes", () => {
+            let a = new BrsString("a");
+            let b = new BrsString("b");
+            let c = new BrsString("c");
+
+            let arr = new BrsArray([a, b, c]);
+
+            arr.set(new Int32(0), new BrsString("replacement for a"));
+            arr.set(new Int32(2), new BrsString("replacement for c"));
+
+            expect(arr.get(new Int32(0))).toEqual(new BrsString("replacement for a"));
+            expect(arr.get(new Int32(2))).toEqual(new BrsString("replacement for c"));
+        });
+
+        it("sets values at out-of-bounds indexes", () => {
```

Comment:
I didn't realize brightscript arrays could do this.  I assume any new indices in between would be undefined?  If so, might be good to add a check for some index like 2xx to verify it's undefined.

Reply:
I could have sworn it was supported but I never actually  checked RBIs behavior. I can do that today – adding bounds checking shouldn’t be too hard if necessary.

Reply:
I was just unaware.  Per the docs: 

> Sets an entry at a given index to the passed value.  If index is beyond the bounds of the array, the array is expanded to accommodate it.

Reply:
Neato! Got a link for that handy?

Reply:
https://sdkdocs.roku.com/display/sdkdoc/ifArraySet#ifArraySet-SetEntry(indexAsInteger,tvalueAsDynamic)AsVoid